backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/ConnectorAsUsageImpl.java

```diff
@@ -228,34 +228,38 @@ public void setOwningRelatedElement(Element newOwningRelatedElement) {
      */
     @Override
     public EList<Element> getRelatedElement() {
-        List<Element> data = new ArrayList<>();
-        return new EcoreEList.UnmodifiableEList<>(this, SysmlPackage.eINSTANCE.getRelationship_RelatedElement(), data.size(), data.toArray());
+        EList<Feature> data = this.getRelatedFeature();
+        return new EcoreEList.UnmodifiableEList(this, SysmlPackage.eINSTANCE.getRelationship_RelatedElement(), data.size(), data.toArray());
     }
 
     /**
-     * <!-- begin-user-doc --> <!-- end-user-doc -->
+     * <!-- begin-user-doc --><!-- end-user-doc -->
      *
-     * @generated
+     * @generated NOT
      */
     @Override
     public EList<Element> getSource() {
-        if (this.source == null) {
-            this.source = new EObjectResolvingEList<>(Element.class, this, SysmlPackage.CONNECTOR_AS_USAGE__SOURCE);
+        EList<Element> sources = new BasicEList<>();
+        Feature sourceFeature = this.getSourceFeature();
+        if (sourceFeature != null) {
+            sources.add(sourceFeature);
         }
-        return this.source;
+        return new EcoreEList.UnmodifiableEList<>(this, SysmlPackage.eINSTANCE.getRelationship_Source(), sources.size(), sources.toArray());
     }
 
     /**
-     * <!-- begin-user-doc --> <!-- end-user-doc -->
+     * <!-- begin-user-doc --><!-- end-user-doc -->
      *
-     * @generated
+     * @generated NOT
```

Comment:
Please a documentation here explaining why the default implementation has been changed and what are the consequences (this list is now unmodifiable)